frontend/src/api/RestaurantApi.tsx

```diff
@@ -1,17 +1,23 @@
+import { SearchState } from '@/pages/SearchPage';
 import { RestaurantResponse } from '@/types';
 import { useAuth0 } from '@auth0/auth0-react';
 import { useQuery } from 'react-query';
 
 const API_BASE_URL = import.meta.env.VITE_API_BASE_URL;
 
-export const useSearchRestaurants = (city?: string) => {
+export const useSearchRestaurants = (
+  searchState: SearchState,
+  city?: string
+) => {
   const { getAccessTokenSilently } = useAuth0();
 
   const createSearchRestaurant = async (): Promise<RestaurantResponse> => {
     const accessToken = await getAccessTokenSilently();
+    const params = new URLSearchParams();
+    params.set('searchQuery', searchState.searchQuery);
 
     const response = await fetch(
-      `${API_BASE_URL}/api/restaurant/search/${city}`,
+      `${API_BASE_URL}/api/restaurant/search/${city}?${params.toString()}`,
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/39#issuecomment-2406781721>#7f995c</a><span> - 10/11/2024, 08:04 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'city' parameter is not encoded before being used in the URL, which could lead to potential security vulnerabilities if it contains special characters or malicious input.
</div>
<div id="fix">
<b>Fix:</b> Encode the 'city' parameter using encodeURIComponent() before including it in the URL to ensure special characters are properly handled and to prevent potential injection attacks.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -19,2 +19,2 @@
      const response = await fetch(
 -      `${API_BASE_URL}/api/restaurant/search/${city}?${params.toString()}`,
 +      `${API_BASE_URL}/api/restaurant/search/${encodeURIComponent(city)}?${params.toString()}`,
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged